Fails to compile under PyPy3.11

```
      creating build/temp.linux-x86_64-pypy311
      gcc -pthread -DNDEBUG -O2 -fPIC -I/usr/include/pypy3.11 -c amsgpack.c -o build/temp.linux-x86_64-pypy311/amsgpack.o -O3 -Werror -Wall -Wextra -Wdouble-promotion -march=native
      amsgpack.c: In function ‘Unpacker_iternext’:
      amsgpack.c:287:23: error: implicit declaration of function ‘_PyDict_NewPresized’ [-Wimplicit-function-declaration]
        287 |       parsed_object = _PyDict_NewPresized(0);
            |                       ^~~~~~~~~~~~~~~~~~~
      amsgpack.c:287:21: error: assignment to ‘PyObject *’ {aka ‘struct _object *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
        287 |       parsed_object = _PyDict_NewPresized(0);
            |                     ^
      amsgpack.c:314:21: error: assignment to ‘PyObject *’ {aka ‘struct _object *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
        314 |       parsed_object = _PyDict_NewPresized(length);
            |                     ^
      amsgpack.c:921:21: error: assignment to ‘PyObject *’ {aka ‘struct _object *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
        921 |       parsed_object = _PyDict_NewPresized(length);
            |                     ^
      error: command '/usr/bin/gcc' failed with exit code 1
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,7 +15,7 @@ jobs:
       security-events: read
     strategy:
       matrix:
-        python-version: ["3.10", "3.12", "3.13"]
+        python-version: ["3.10", "3.12", "3.13", "pypy3.11"]
         compiler: ["gcc", "clang"]
     steps:
       - uses: actions/checkout@v4

--- a/amsgpack.c
+++ b/amsgpack.c
@@ -27,6 +27,12 @@ static PyMethodDef AMsgPackMethods[] = {
     {NULL, NULL, 0, NULL}  // Sentinel
 };
 
+#ifndef PYPY_VERSION
+#define ANEW_DICT _PyDict_NewPresized
+#else
+#define ANEW_DICT(N) PyDict_New()
+#endif
+
 static struct PyModuleDef amsgpack_module = {.m_base = PyModuleDef_HEAD_INIT,
                                              .m_name = "amsgpack",
                                              .m_doc = NULL,
@@ -284,7 +290,7 @@ parse_next:
   PyObject* parsed_object = NULL;
   switch (next_byte) {
     case '\x80': {
-      parsed_object = _PyDict_NewPresized(0);
+      parsed_object = ANEW_DICT(0);
       if A_UNLIKELY(parsed_object == NULL) {
         return NULL;
       }
@@ -311,7 +317,7 @@ parse_next:
         return NULL;
       }
       Py_ssize_t const length = Py_CHARMASK(next_byte) & 0x0f;
-      parsed_object = _PyDict_NewPresized(length);
+      parsed_object = ANEW_DICT(length);
       if A_UNLIKELY(parsed_object == NULL) {
         return NULL;
       }
@@ -918,7 +924,7 @@ parse_next:
         PyErr_SetString(PyExc_ValueError, "Deeply nested object");
         return NULL;
       }
-      parsed_object = _PyDict_NewPresized(length);
+      parsed_object = ANEW_DICT(length);
       if (length == 0) {
         break;
       }

--- a/test/failing_malloc.py
+++ b/test/failing_malloc.py
@@ -2,20 +2,22 @@ from __future__ import annotations
 from typing import Literal
 
 # import ctypes
+try:
+    from ctypes import pythonapi
+except ImportError:  # pypy
+    pythonapi = None
+AVAILABLE = pythonapi is not None
 from ctypes import (
     byref,
     c_size_t,
     c_void_p,
     cast as c_cast,
-    cdll,
     CFUNCTYPE,
     POINTER,
     pointer as c_pointer,
-    pythonapi,
     Structure as CStructure,
 )
 from contextlib import contextmanager
-from pathlib import Path
 
 # PYMEM_DOMAIN_RAW = 0
 # PYMEM_DOMAIN_MEM = 1

--- a/test/test_pack.py
+++ b/test/test_pack.py
@@ -1,8 +1,9 @@
+from unittest import skipUnless
 from math import pi
 from amsgpack import packb, Ext, unpackb
 from struct import pack
 from .test_amsgpack import SequenceTestCase
-from .failing_malloc import failing_malloc
+from .failing_malloc import failing_malloc, AVAILABLE as FAILING_AVAILABLE
 
 
 class PackbTest(SequenceTestCase):
@@ -155,6 +156,7 @@ class PackbTest(SequenceTestCase):
             packb(outer)
         self.assertEqual(str(context.exception), "Deeply nested object")
 
+    @skipUnless(FAILING_AVAILABLE, "not failing available")
     def test_initial_buffer_failure(self):
         with self.assertRaises(MemoryError), failing_malloc(1023, "raw"):
             packb(0)
@@ -164,7 +166,10 @@ class PackbIntTest(SequenceTestCase):
     def test_out_of_range(self):
         with self.assertRaises(OverflowError) as context:
             packb(0x1_FFFF_FFFFF_FFFF_FFFF)
-        self.assertEqual(str(context.exception), "int too big to convert")
+        self.assertIn(
+            str(context.exception),
+            ("int too big to convert", "integer too large"),
+        )
 
     def test_i8(self):
         for ref in range(-0x80, -0x20):

--- a/test/test_unpack.py
+++ b/test/test_unpack.py
@@ -1,6 +1,7 @@
 from amsgpack import packb, Unpacker, Ext, unpackb, Raw
-from .failing_malloc import failing_malloc
+from .failing_malloc import failing_malloc, AVAILABLE as FAILING_AVAILABLE
 from .test_amsgpack import SequenceTestCase
+from unittest import skipUnless
 
 
 class UnpackerTest(SequenceTestCase):
@@ -129,6 +130,7 @@ class UnpackerTest(SequenceTestCase):
             u.feed(bytes((char,)))
         self.safeSequenceEqual(u, (b"\x03\x04",))
 
+    @skipUnless(FAILING_AVAILABLE, "not failing available")
     def test_bin_malloc_failure(self):
         u = Unpacker()
         for char in b"\xc4\xffaaaaa":
@@ -188,6 +190,7 @@ class UnpackerTest(SequenceTestCase):
             u.feed(seq)
         self.assertEqual(list(u), [124, "", 0])
 
+    @skipUnless(FAILING_AVAILABLE, "not failing available")
     def test_bin_PyBytes_FromStringAndSize_malloc(self):
         u = Unpacker()
         u.feed(b"\xc6\x00\x01\x00\x00" + b"A" * 0x10000)
@@ -200,6 +203,7 @@ class UnpackerTest(SequenceTestCase):
         with self.assertRaises(StopIteration):
             next(u)
 
+    @skipUnless(FAILING_AVAILABLE, "not failing available")
     def test_feed_no_memory(self):
         u = Unpacker()
         with self.assertRaises(MemoryError), failing_malloc(9, "mem"):

